tests/components/google_wifi/test_sensor.py

```diff
@@ -34,185 +34,204 @@
+    resource = f"http://localhost{google_wifi.ENDPOINT}"
+    requests_mock.Mocker().get(resource, status_code=200)
+    assert await async_setup_component(
+        hass,
+        "sensor",
+        {
+            "sensor": {
+                "platform": "google_wifi",
+                "host": "localhost",
+                "name": "Test Wifi",
+                "monitored_conditions": [
+                    "current_version",
+                    "new_version",
+                    "uptime",
+                    "last_restart",
+                    "local_ip",
+                    "status",
+                ],
             }
+        },
+    )
+    assert_setup_component(6, "sensor")
 
-    def fake_delay(self, ha_delay):
-        """Fake delay to prevent update throttle."""
-        hass_now = dt_util.utcnow()
-        shifted_time = hass_now + timedelta(seconds=ha_delay)
-        self.hass.bus.fire(ha.EVENT_TIME_CHANGED, {ha.ATTR_NOW: shifted_time})
 
-    def test_name(self):
-        """Test the name."""
-        for name in self.sensor_dict:
-            sensor = self.sensor_dict[name]["sensor"]
-            test_name = self.sensor_dict[name]["name"]
-            assert test_name == sensor.name
+"""Tests for Google Wifi sensor platform."""
 
-    def test_unit_of_measurement(self):
-        """Test the unit of measurement."""
-        for name in self.sensor_dict:
-            sensor = self.sensor_dict[name]["sensor"]
-            assert self.sensor_dict[name]["units"] == sensor.unit_of_measurement
 
-    def test_icon(self):
-        """Test the icon."""
+@pytest.fixture
+def setUp(self):
+    """Set up things to be run when tests are started."""
+    with requests_mock.Mocker() as mock_req:
+        self.setup_api(MOCK_DATA, mock_req)
+    return self.addCleanup(self.hass.stop)
+
+
+def setup_api(self, data, mock_req):
+    """Set up API with fake data."""
+    resource = f"http://localhost{google_wifi.ENDPOINT}"
+    now = datetime(1970, month=1, day=1)
+    with patch("homeassistant.util.dt.now", return_value=now):
+        mock_req.get(resource, text=data, status_code=200)
+        conditions = google_wifi.MONITORED_CONDITIONS.keys()
+        self.api = google_wifi.GoogleWifiAPI("localhost", conditions)
+    self.name = NAME
+    self.sensor_dict = {}
+    for condition, cond_list in google_wifi.MONITORED_CONDITIONS.items():
+        sensor = google_wifi.GoogleWifiSensor(self.api, self.name, condition)
+        name = f"{self.name}_{condition}"
+        units = cond_list[1]
+        icon = cond_list[2]
+        self.sensor_dict[condition] = {
+            "sensor": sensor,
+            "name": name,
+            "units": units,
+            "icon": icon,
+        }
+
+
+def fake_delay(self, ha_delay):
+    """Fake delay to prevent update throttle."""
+    hass_now = dt_util.utcnow()
+    shifted_time = hass_now + timedelta(seconds=ha_delay)
+    self.hass.bus.fire(ha.EVENT_TIME_CHANGED, {ha.ATTR_NOW: shifted_time})
+
+
+@pytest.fixture
```

Comment:
This makes no sense. Now it's not executing this test. Remove fixture from all the tests.